cbmc/proofs/IotMqtt_PublishAsync/IotMqtt_PublishAsync_harness.c

```diff
@@ -0,0 +1,79 @@
+#include "iot_config.h"
+#include "private/iot_mqtt_internal.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "mqtt_state.h"
+
+/**
+ * We abstract all functions related to concurrency and assume they are correct.
+ * Thus, we add these stub to increase coverage.
+ */
+IotTaskPoolError_t IotTaskPool_CreateJob( IotTaskPoolRoutine_t userCallback,
+                                          void * pUserContext,
+                                          IotTaskPoolJobStorage_t * const pJobStorage,
+                                          IotTaskPoolJob_t * const pJob )
+{
+  /* _IotMqtt_ScheduleOperation asserts this */
+  return IOT_TASKPOOL_SUCCESS;
+}
+
+IotTaskPoolError_t IotTaskPool_ScheduleDeferred( IotTaskPool_t taskPool,
+                                                 IotTaskPoolJob_t job,
+                                                 uint32_t timeMs )
```

Comment:
Any validity checks to do on these types?

Reply:
This function also act if the parameters are NULL. It checks them with `TASKPOOL_ON_NULL_ARG_GOTO_CLEANUP`, which exit if an argument is NULL.

Reply:
No, we shouldn't. In this case, `_IotMqtt_ScheduleOperation` asserts that `IotTaskPool_ScheduleDeferred` will not return a `IOT_TASKPOOL_BAD_PARAMETER`; thus, I must assume it in the proof as well.

Reply:
OK